src/target/riscv/riscv-011.c

```diff
@@ -455,6 +455,9 @@ static uint64_t dbus_read(struct target *target, uint16_t address)
 	dbus_status_t status;
 	uint16_t address_in;
 
+	// First, discard stale data
```

Comment:
This comment is a bit misleading... you aren't JUST discarding stale data... you're also actually sending the address that you want to read.